### ISSUE TITLE
Suppress doesn't appear to work with id

### ISSUE THREAD

**Issue by leventov** [checkstyle/checkstyle#6181]
When I have a suppression config like this:
```xml
<suppressions>
  <suppress checks="Regexp" id="someId" files="[\\/]src[\\/]test[\\/]" />
</suppressions>
```
Checkstyle fails with the following error:
```
Attribute "id" must be declared for element type "suppress".
```
If I remove the "checks" property, then
```
Attribute "checks" is required and must be specified for element type "suppress".
```

It doesn't seem to work as described here: http://checkstyle.sourceforge.net/config_filters.html#SuppressionFilter

**Comment by rnveach** [comment]
@leventov Please see http://checkstyle.sourceforge.net/report_issue.html#How_to_report_a_bug.3F .

I am unable to reproduce your problem. The only error messages checkstyle generates for these, that I know of, are https://github.com/checkstyle/checkstyle/blob/ecd1b65bd63546d1245b6415759c3c2d1e2b77a4/src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressionsLoader.java#L163 and https://github.com/checkstyle/checkstyle/blob/ecd1b65bd63546d1245b6415759c3c2d1e2b77a4/src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressionsLoader.java#L174 .

**Comment by romani** [comment]
@leventov , please always follow issue template, it will give us all details in one iteration (to avoid multiple iterations in issue discussion). In many case it force user understand that he is doing smth wrong :).

**Comment by rnveach** [comment]
@leventov 
https://github.com/metamx/druid/commit/fd641fbdf1063570c8f9a41bbb033a1337eb9a20#diff-1adfe9776419c5bd4293259e36d15babL53

When you upgraded checkstyle, you didn't upgrade suppression DTD. Please try updating it and see if your problem still continues.

**Comment by rnveach** [comment]
@leventov It is the DTD validator that created the error message. Checkstyle didn't produce it and I am not sure if there is anyway to fine tune it.

Here is the full error message.
````
com.puppycrawl.tools.checkstyle.api.CheckstyleException: cannot initialize module SuppressionFilter - Unable to parse C:\D\Rickys\Java\checkstyleWorkspaceEclipse\checkstyle\config\suppressions.xml - Attribute "id" must be declared for element type "suppress".
	at com.puppycrawl.tools.checkstyle.Checker.setupChild(Checker.java:459)
	at com.puppycrawl.tools.checkstyle.api.AutomaticBean.configure(AutomaticBean.java:198)
	at com.puppycrawl.tools.checkstyle.Main.runCheckstyle(Main.java:577)
	at com.puppycrawl.tools.checkstyle.Main.runCli(Main.java:472)
	at com.puppycrawl.tools.checkstyle.Main.main(Main.java:226)
Caused by: com.puppycrawl.tools.checkstyle.api.CheckstyleException: Unable to parse C:\checkstyleWorkspaceEclipse\checkstyle\config\suppressions.xml - Attribute "id" must be declared for element type "suppress".
	at com.puppycrawl.tools.checkstyle.filters.SuppressionsLoader.getSuppressionLoader(SuppressionsLoader.java:283)
	at com.puppycrawl.tools.checkstyle.filters.SuppressionsLoader.loadSuppressions(SuppressionsLoader.java:232)
	at com.puppycrawl.tools.checkstyle.filters.SuppressionsLoader.loadSuppressions(SuppressionsLoader.java:219)
	at com.puppycrawl.tools.checkstyle.filters.SuppressionFilter.finishLocalSetup(SuppressionFilter.java:100)
	at com.puppycrawl.tools.checkstyle.api.AutomaticBean.configure(AutomaticBean.java:194)
	at com.puppycrawl.tools.checkstyle.Checker.setupChild(Checker.java:455)
	... 4 more
Caused by: org.xml.sax.SAXParseException; systemId: file:/C:/D/Rickys/Java/checkstyleWorkspaceEclipse/checkstyle/config/suppressions.xml; lineNumber: 8; columnNumber: 74; Attribute "id" must be declared for element type "suppress".
	at com.sun.org.apache.xerces.internal.util.ErrorHandlerWrapper.createSAXParseException(ErrorHandlerWrapper.java:203)
	at com.sun.org.apache.xerces.internal.util.ErrorHandlerWrapper.error(ErrorHandlerWrapper.java:134)
	at com.sun.org.apache.xerces.internal.impl.XMLErrorReporter.reportError(XMLErrorReporter.java:396)
	at com.sun.org.apache.xerces.internal.impl.XMLErrorReporter.reportError(XMLErrorReporter.java:327)
	at com.sun.org.apache.xerces.internal.impl.XMLErrorReporter.reportError(XMLErrorReporter.java:284)
	at com.sun.org.apache.xerces.internal.impl.dtd.XMLDTDValidator.addDTDDefaultAttrsAndValidate(XMLDTDValidator.java:1253)
	at com.sun.org.apache.xerces.internal.impl.dtd.XMLDTDValidator.handleStartElement(XMLDTDValidator.java:1917)
	at com.sun.org.apache.xerces.internal.impl.dtd.XMLDTDValidator.emptyElement(XMLDTDValidator.java:763)
	at com.sun.org.apache.xerces.internal.impl.XMLNSDocumentScannerImpl.scanStartElement(XMLNSDocumentScannerImpl.java:357)
	at com.sun.org.apache.xerces.internal.impl.XMLDocumentFragmentScannerImpl$FragmentContentDriver.next(XMLDocumentFragmentScannerImpl.java:2787)
	at com.sun.org.apache.xerces.internal.impl.XMLDocumentScannerImpl.next(XMLDocumentScannerImpl.java:606)
	at com.sun.org.apache.xerces.internal.impl.XMLNSDocumentScannerImpl.next(XMLNSDocumentScannerImpl.java:118)
	at com.sun.org.apache.xerces.internal.impl.XMLDocumentFragmentScannerImpl.scanDocument(XMLDocumentFragmentScannerImpl.java:510)
	at com.sun.org.apache.xerces.internal.parsers.XML11Configuration.parse(XML11Configuration.java:848)
	at com.sun.org.apache.xerces.internal.parsers.XML11Configuration.parse(XML11Configuration.java:777)
	at com.sun.org.apache.xerces.internal.parsers.XMLParser.parse(XMLParser.java:141)
	at com.sun.org.apache.xerces.internal.parsers.AbstractSAXParser.parse(AbstractSAXParser.java:1213)
	at com.sun.org.apache.xerces.internal.jaxp.SAXParserImpl$JAXPSAXParser.parse(SAXParserImpl.java:643)
	at com.puppycrawl.tools.checkstyle.XmlLoader.parseInputSource(XmlLoader.java:85)
	at com.puppycrawl.tools.checkstyle.filters.SuppressionsLoader.getSuppressionLoader(SuppressionsLoader.java:274)
	... 9 more
Checkstyle ends with 1 errors.
````

@romani Any thoughts?

**Comment by romani** [comment]
>Caused by: org.xml.sax.SAXParseException; systemId: file:/C:/D/Rickys/Java/checkstyleWorkspaceEclipse/checkstyle/config/suppressions.xml; lineNumber: 8; columnNumber: 74; Attribute "id" must be declared for element type "suppress".

I doubt that we can do message better. But yes it demand from user to understand what is xml , and what is schema/dtd .

In old schema there is no `id`

https://github.com/checkstyle/checkstyle/blob/master/src/main/resources/com/puppycrawl/tools/checkstyle/suppressions_1_0.dtd

**Comment by alailsonbraz** [comment]
Try run checkstyle with maven.

mvn checkstyle:check -U
